### DISCUSSION TITLE
vi motions with visual mode over command line

### DISCUSSION THREAD

**Discussion by yriveiro** [wezterm/wezterm#6390] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I'm trying to emulate the TMUX plugin [zsh-vi-mode](https://github.com/jeffreytse/zsh-vi-mode) using Wezterm functionalities.

I found that there is too much overlap between TMUX and Wezterm. I achieved almost everything I needed with Wezterm but I've hit a roadblock.

Normally, I use this mode to navigate the command line and change stuff. I tried building a new key table with all the keybindings and for movement and positioning of the cursor, everything works. My problem is when I want to do a simple Vim motion to select the text and, in the end, delete it.

Did someone do anything remotely close to what I'm describing? Is it even possible?

**Comment by bew** [comment, marked as answer]
Note that zsh-vi-mode is a zsh plugin, not a tmux plugin. This is important because it has access to zsh internals and can request zsh to highlight text (to be deleted).
From the terminal' point of view there is no standard way to tell the shell to select some text do be deleted later, so I'm afraid what you're trying to do isn't really possible 😬

**Comment by yriveiro** [comment]
> Note that zsh-vi-mode is a zsh plugin, not a tmux plugin

🎯 on that.

In my head, the rationale was that this new mode would be a copy mode with only horizontal movement doing the job, but as you mention, that requires zsh integration, and it is not the mission of Wezterm.

I tried to have two different configurations activating copy mode (one full and another with restricted horizontal move), but once I call `ActivateCopyMode`, the keybindings get funky.

Thanks for the clarification @bew
